public/app/plugins/datasource/cloudwatch/components/MetricsQueryEditor.tsx

```diff
@@ -153,21 +163,20 @@ export class MetricsQueryEditor extends PureComponent<Props, State> {
             <label className="gf-form-label">
               <a
                 onClick={() =>
-                  executedQueries &&
+                  executedQueryPreview &&
                   this.setState({
                     showMeta: !showMeta,
                   })
                 }
               >
-                <Icon name={showMeta && executedQueries ? 'angle-down' : 'angle-right'} />{' '}
-                {showMeta && executedQueries ? 'Hide' : 'Show'} Query Preview
+                <Icon name={showMeta ? 'angle-down' : 'angle-right'} /> {showMeta ? 'Hide' : 'Show'} Query Preview
               </a>
             </label>
           </div>
           <div className="gf-form gf-form--grow">
             <div className="gf-form-label gf-form-label--grow" />
           </div>
-          {showMeta && executedQueries && (
+          {showMeta && (
```

Comment:
you removed the `&& executedQueryPreview` safeguard on purpose?

Reply:
yes this checked is passed to getExecutedQueryPreview